saleor/graphql/product/mutations/products.py

```diff
@@ -1784,6 +1784,67 @@ def perform_mutation(cls, _root, info, product_id, images_ids):
         return ProductImageReorder(product=product, images=images)
 
 
+class ProductVariantReorder(BaseMutation):
+    product = graphene.Field(Product)
+    variants = graphene.List(ProductVariant)
+
+    class Arguments:
+        product_id = graphene.ID(
+            required=True,
+            description="Id of product that variants order will be altered.",
+        )
+        variants = graphene.List(
+            graphene.ID,
+            required=True,
+            description="IDs of a product variants in the desired order.",
+        )
+
+    class Meta:
+        description = "Changes ordering of the product variants."
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, product_id, variants):
+        product = cls.get_node_or_error(
+            info, product_id, field="product_id", only_type=Product
+        )
+
+        if len(variants) != product.variants.count():
+            raise ValidationError(
+                {
+                    "order": ValidationError(
+                        "Incorrect number of variant IDs provided.",
+                        code=ProductErrorCode.INVALID,
+                    )
+                }
+            )
+
+        ordered_variants = []
+        for variant_id in variants:
+            variant = cls.get_node_or_error(
+                info, variant_id, field="order", only_type=ProductVariant
+            )
+            if variant and variant.product != product:
```

Comment:
IMO This will trigger additional DB queries, I don't think that we fetched product:
```
In [34]: from saleor.product.models import ProductVariant

In [35]: v = ProductVariant.objects.all()

In [36]: for x in v:
    ...:     w = x.product.name
    ...:
Number of queries 130

In [37]: v = ProductVariant.objects.select_related('product').all()

In [38]: for x in v:
    ...:     w = x.product.name
    ...:
Number of queries 1
```

Reply:
Yeah, copied that from other ordering mutations but this can be optimised further, get rid of `for` loop and use `get_nodeS_or_error`, will do that in a moment

Reply:
Please don't use `select_related` for sparse many-to-one relations (eg. when many variants can map to a single product) as it will result in the database sending us multiple copies of the same related row. `prefetch_related` is often a better idea when the relation is not one-to-one.